File: Restpirators.DataAccess/Entities/Emergency.cs
using Restpirators.Common.Enums;
using System;

namespace Restpirators.DataAccess.Entities
{
    public class Emergency : BaseEntity
    {
        public int EmergencyTypeId { get; set; }
        public virtual EmergencyType EmergencyType { get; set; }
        public string Location { get; set; }
        public string Description { get; set; }
        public DateTime ReportTime { get; set; }
        public EmergencyStatus Status { get; set; }
        public string ModUser { get; set; }
        public int? AssignedToTeamId { get; set; }
        public virtual Team AssignedToTeam { get; set; }
        public string Identifier { get; set; }
    }
}

File: emergency-web-client/src/app/_model/Emergency.ts
export class Emergency {
  Id: number;
  emergencyTypeId: number;
  location: string;
  description: string;
  reportTime: Date;
  status: number;
  modUser: string;
  identifier: string;

  constructor(type: number, location, description, identifier) {
    this.emergencyTypeId = type;
    this.location = location;
    this.description = description;
    this.reportTime = new Date();
    this.identifier = identifier;
  }
}
